### DISCUSSION TITLE
[Feature] Include album ids (if any) in searchMetadata api

### DISCUSSION THREAD

**Discussion by schrauger** [immich-app/immich#11858] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When querying the searchMetadata api endpoint, I would like the list of image details to include an array of the albums that photo appears in. Right now, that information isn't listed in the photo list.

Since there is already a parameter "isNotInAlbum" (which limits photos not in any album), it would seem the knowledge of what albums contain that photo already exists somewhere. It would be nice to have the list of album ids included, whether an empty array for photos that aren't in an album, or an array with the album ids that it's in.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by schrauger** [comment]
Ultimately, I want to use this info to search for a specific person and iterate through the list of photos to find what albums that person appears in. Right now, I'd need to get a list of all the photos, then query the api for each photo to get any album info. 

Though even the getAssetInfo endpoint doesn't seem to list album information.

**Comment by schuhbacca** [comment]
Guessing it's probably for speed. Search doesn't display anything related to albums, so no need to slow the query down by joining on an unnecessary table.

**Comment by mertalev** [comment]
It's a bit trickier to query for this without causing duplicate entries in the results. In fact, I suspect there's a bug related to this for the "not in album" filter as well.

We currently have a PoC that makes big changes to how we query the database. Among other things, it will make it easier to implement more sophisticated queries to enable these kinds of features.
